### PR TITLE
Fix Inline/G2Invoke Problem  

### PR DESCRIPTION
**Pull request by calebmkim**
Addresses #1283. As @rachitnigam alluded to in the final comment of that thread, the problem was that we were only rewriting invokes of components we are going to inline. However, there are times when we need to rewrite invokes for components that aren't being inlined (see the test case for an example). I'm not sure if I have handled this in the cleanest way, so if there's a cleaner way to do this lmk.

### CODE REVIEW

File: calyx/src/passes/component_iniliner.rs [hunk 1]
```diff
@@ -122,6 +122,30 @@ impl ComponentInliner {
         })
     }
 
+    /// Rewrites vec based on self.interface_rewrites
+    /// Used as a helper function for rewrite_invoke_ports
+    fn rewrite_vec(
+        &self,
+        v: &mut Vec<(ir::Id, RRC<ir::Port>)>,
+    ) -> Vec<(ir::Id, RRC<ir::Port>)> {
+        v.drain(..)
```
Comment (rachitnigam) [inline calyx/src/passes/component_iniliner.rs:131]: Consider using iter_mut here instead! https://doc.rust-lang.org/std/primitive.slice.html#method.iter_mut

**Comment by rachitnigam** [comment]
Ah, makes perfect sense! Left a minor comment about using a different iteration method. Merge at your discretion

**Comment by rachitnigam** [comment]
Merging this myself because I'm going to make some changes to the inliner soon

**Comment by sampsyo** [comment]
Nice; this looks like a simple solution to me!

**Pull request merged**